### ISSUE TITLE
While running it in windows. select.poll() is not working.

### ISSUE THREAD

**Issue by rsmrahul1** [andreikop/python-ws-discovery#1]
While running it in windows. select.poll() is not working.
Any work around ?

**Comment by petri** [comment]
@rsmrahul1 see #4 and the fork by @tran-wang.

**Comment by lafernando** [comment]
Great to see you guys have continued the project with bug fixes and all :) .. After the initial project done in google code, I couldn't really look into it. I did the project as part of a project in my undergraduate degree. Glad to see it is useful to others too.

**Comment by petri** [comment]
I suggest we close this. The package claims no Windows support (in trove classifiers), and the forks that support Windows, merely seem to drop the functionality that uses `select.poll`, rather than implementing  any Windows-specific replacement.

**Comment by andreikop** [comment]
I keep it open to
1. avoid reopening by new reporters
2. with hope that somebody will fix it sometimes
But, I don't mind closing.

**Comment by petri** [comment]
Wouldn't avoiding new "Windows support" issues be best handled by documentation? The package is not really explicit about that at the moment, which imho is why this has happened.

As for adding better Windows support, looks like it might be possible to take advantage of pywin32 IOCP support. The pycos package (https://pypi.org/project/pycos/) uses it internally and might thus be a good place to start from. Based on some googling, this is pretty much the only way to go. 

But on the other hand, if we could decouple the threaded networking implementation from the rest of the code and then add asyncio support, that would "fix" the issue, too.

**Comment by andreikop** [comment]
PR which adds to README.md a section with explanation of Windows problems and ideas how some contributor can fix this problems is welcome!

**Comment by rjsdotorg** [comment]
> PR which adds to README.md a section with explanation of Windows problems and ideas how some contributor can fix this problems is welcome!

I ported to Python 3.8 for Windows 7+ 
NOT tested on POSIX systems
numerous changes to fix deprecations in Py38

[WSDiscovery.zip](https://github.com/andreikop/python-ws-discovery/files/4360670/WSDiscovery.zip)

**Comment by rjsdotorg** [comment]
> PR which adds to README.md a section with explanation of Windows problems and ideas how some contributor can fix this problems is welcome!

I ported to Python 3.8 for Windows 7+
NOT tested on POSIX systems
numerous changes to fix deprecations in Py38
[WSDiscovery.zip](https://github.com/andreikop/python-ws-discovery/files/4361013/WSDiscovery.zip)

**Comment by petri** [comment]
Python 3.4 introduced [selectors](https://docs.python.org/3/library/selectors.html) that abstracts away the underlying OS-specific implementation.  The docs indicate that this might work on Windows too.

I'm on a Mac myself (no `select.poll` here neither) so I went & replaced `select.poll` based threading with selectors. Result here in a feature branch (based on the current 2.x development version): https://github.com/koodaamo/python-ws-discovery/tree/selectors . 

I updated the simple Probe test too and things look alright. If someone here is on Windows, try it out & let me know if it works for you.
